### PR TITLE
Fix warnings and compile errors on nightly

### PR DESCRIPTION
**Pull request by arpad-m**
Nightly has added a bunch of compiler and linter warnings. There is also two dependencies that fail compilation on latest nightly due to using the old `stdsimd` feature name. This PR fixes them.

### CODE REVIEW

**Review by conradludgate — APPROVED** [review]

File: libs/utils/src/completion.rs [hunk 1]
```diff
@@ -49,5 +51,5 @@ pub fn channel() -> (Completion, Barrier) {
     tracker.close();
 
     let token = tracker.token();
-    (Completion(token), Barrier(tracker))
```
Comment (koivunej) [inline libs/utils/src/completion.rs:52]: Why this?

File: safekeeper/src/handler.rs [hunk 2]
```diff
@@ -30,7 +29,7 @@ use utils::{
 /// Safekeeper handler of postgres commands
 pub struct SafekeeperPostgresHandler {
     pub conf: SafeKeeperConf,
-    /// assigned application name
+    /// assigned applicatioun name
```
Comment (koivunej) [inline safekeeper/src/handler.rs:32]:
```suggestion
    /// assigned application name
```

Old one was fine :)

File: libs/utils/src/completion.rs (hunk 1 above)
Comment (koivunej) [reply libs/utils/src/completion.rs:52]: Aah, better no one reads the field, interesting.

**Review by koivunej — APPROVED** [review]

File: safekeeper/src/handler.rs (hunk 2 above)
Comment (arpad-m) [reply safekeeper/src/handler.rs]: seems to have been some typo, fixed.

**Review by petuhovskiy — APPROVED** [review]

File: libs/utils/src/completion.rs (hunk 1 above)
Comment (koivunej) [reply libs/utils/src/completion.rs:52]: Unsure how I feel about this warning for single field  newtypes. Doing exactly this is a very valid pattern... I would hope they'd check for drop in the lint before stabilizing this.

**Pull request merged**